lib/utils/stream_extension.dart

```diff
@@ -45,4 +47,44 @@ extension StreamExtension on Stream {
     };
     return controller.stream;
   }
+
+  Stream<SyncUpdate> rateLimitWithSyncUpdate(Duration t) {
+    final controller = StreamController<SyncUpdate>();
+    Timer? timer;
+    SyncUpdate? pendingMessage;
+
+    void processMessage() {
+      if (controller.isClosed) return;
+
+      if (timer == null) {
+        if (pendingMessage != null) {
+          controller.add(pendingMessage!);
+          pendingMessage = null;
+          timer = Timer(t, () {
+            timer = null;
+            if (pendingMessage != null) {
```

Comment:
Here pendingMessage is null since you changed its value line 62
https://github.com/linagora/twake-on-matrix/blob/014414586c9babc75bf0d490bb93f3a0b169040d/lib/utils/stream_extension.dart#L62